ui/controlgroup.js

```diff
@@ -0,0 +1,182 @@
+/*!
+ * jQuery UI Controlgroup @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright 2014 jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ *
+ * http://api.jqueryui.com/controlgroup/
+ */
+( function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define( [
+			"jquery",
+			"./core",
+			"./widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}( function( $ ) {
+
+return $.widget( "ui.controlgroup", {
+	version: "@VERSION",
+	defaultElement: "<div>",
+	options: {
+		disabled: null,
+		items: {
+			"button": "input[type=button], input[type=submit], input[type=reset], button, a",
+			"checkboxradio": "input[type='checkbox'], input[type='radio']",
+			"selectmenu": "select"
+		},
+		direction: "horizontal",
+		excludeInvisible: true,
+		classes: {}
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// The support the enhanced option in jQuery Mobile, we isolate DOM manipulation
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map( function() {
+				return $( this )[ widget ]( "widget" ).removeData( "ui-controlgroup-data" )[ 0 ];
+			} ).removeData( "ui-controlgroup-data" );
+		} );
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+
+		this.buttons = $();
```

Comment:
Because right now it only supports "button like widgets" however this will not be true as soon as we add textinput widget so i should probably change the name. Problem is this is not necessarily refereeing to the actual widget elements but rather the visual representation of it. not sure what the best name is. It does represent the result of the `widget` method of each though so maybe `childWidgets` is best?